### PR TITLE
Merge slovdahl to moimael

### PR DESCRIPTION
**Pull request by jfkw**
Please pull to integrate @slovdahl changes into moimael master. Commits created by cherry-pick of individual commits from slovdahl/master after common ancestor. No changes made to @slovdahl commits other than merge conflict resolution.

### CODE REVIEW

File: migrate.py [hunk 1]
```diff
@@ -37,42 +38,44 @@
 
 """
 
+reload(sys)
```
Comment (moimael) [inline migrate.py]: I'm not super fond of this:

http://stackoverflow.com/questions/3828723/why-should-we-not-use-sys-setdefaultencodingutf-8-in-a-py-script

**Review by moimael — CHANGES_REQUESTED** [review]

**Comment by slovdahl** [comment]
I had no intention of getting this merged upstream myself, it's just a collection of random stuff I had to change/fix to make the script work for our specific migration.

**Comment by slovdahl** [comment]
Would it be a better idea to submit a few separate pull requests with specific changes? GitLab 8.15 database scheme, improved trac wiki syntax conversion, etc.

**Comment by moimael** [comment]
Yes I understand, it's an issue with this kind of project, it's used once so nobody want to maintain it. I guess I can accept the PR and if someone want to clean it up, he can do it.

**Comment by moimael** [comment]
Yes maybe separate PR would be better, but does your changes still work without the utf8 hack ?

**Comment by slovdahl** [comment]
The utf8 hack was needed to migrate some ticket descriptions and comments that contained obscure characters, so I think generally, yes, it should work without it.

**Comment by jfkw** [comment]
@slovdahl apologies if I brought over commits you didn't intend. I went to look for git private messaging to check with you first about it, but they've removed that channel.

I added a commit to revert setdefaultencoding('utf-8').

@slovdahl I left in the encoding argument to xmlrpc.ServerProxy(encoding='utf-8'), is this sufficient to handle the special characters you encountered?

**Comment by slovdahl** [comment]
@jfkw No need to apologise! If I didn't want to risk this I wouldn't have made it public on GitHub at all :) Just didn't intend to do it myself.

I'm cleaning up the repo a little bit at the moment and will submit separate pull requests for at least some of the issues.

I'm unsure about the utf8 issue and I'm unfortunately not able to verify it at the moment either.

**Comment by jfkw** [comment]
Don't mind my updates to the PR branch. It sounds like @slovdahl is going to submit mulitple PRs to merge changes. I'm just trying to keep my branch in a merge-ready state while I move forward.

**Comment by slovdahl** [comment]
@jfkw would you mind rebasing this branch on top of the current master and check if there's anything else you're interested in having merged upstream? I'd be happy to submit further pull requests in that case.

**Comment by jfkw** [comment]
@slovdahl Yes, I'll redo the branch based on current head.

**Pull request closed without merging**